discord/sku.py

```diff
@@ -0,0 +1,175 @@
+from __future__ import annotations
+
+from typing import Optional, TYPE_CHECKING
+
+from . import utils
+from .app_commands import MissingApplicationID
+from .enums import try_enum, SKUType, EntitlementType
+from .flags import SKUFlags
+
+if TYPE_CHECKING:
+    from datetime import datetime
+
+    from .guild import Guild
+    from .state import ConnectionState
+    from .types.sku import (
+        SKU as SKUPayload,
+        Entitlement as EntitlementPayload,
+    )
+    from .user import User
+
+__all__ = (
+    'SKU',
+    'Entitlement',
+)
+
+
+class SKU:
+    """Represents a premium offering as a stock-keeping unit (SKU).
+
+    .. versionadded:: 2.4
+
+    Attributes
+    -----------
+    id: :class:`int`
+        The SKU's ID.
+    type: :class:`SKUType`
+        The type of the SKU.
+    application_id: :class:`int`
+        The ID of the application that the SKU belongs to.
+    name: :class:`str`
+        The consumer-facing name of the premium offering.
+    slug: :class:`str`
+        A system-generated URL slug based on the SKU name.
+    """
+
+    __slots__ = (
+        '_state',
+        'id',
+        'type',
+        'application_id',
+        'name',
+        'slug',
+        '_flags',
+    )
+
+    def __init__(self, *, state: ConnectionState, data: SKUPayload):
+        self._state: ConnectionState = state
+        self.id: int = int(data['id'])
+        self.type: SKUType = try_enum(SKUType, data['type'])
+        self.application_id: int = int(data['application_id'])
+        self.name: str = data['name']
+        self.slug: str = data['slug']
+        self._flags: int = data['flags']
+
+    def __repr__(self) -> str:
+        return f'<SKU id={self.id} name={self.name!r} slug={self.slug!r}>'
+
+    @property
+    def flags(self) -> SKUFlags:
+        """Returns the flags of the SKU."""
+        return SKUFlags._from_value(self._flags)
+
+    @property
+    def created_at(self) -> datetime:
+        """:class:`datetime.datetime`: Returns the sku's creation time in UTC."""
+        return utils.snowflake_time(self.id)
+
+
+class Entitlement:
+    """Represents an entitlement from user or guild which has been granted access to a premium offering.
+
+    .. versionadded:: 2.4
+
+    Attributes
+    -----------
+    id: :class:`int`
+        The entitlement's ID.
+    sku_id: :class:`int`
+        The ID of the SKU that the entitlement belongs to.
+    user_id: Optional[:class:`int`]
+        The ID of the user that is granted access to the entitlement.
+    guild_id: Optional[:class:`int`]
+        The ID of the guild that is granted access to the entitlement
+    application_id: :class:`int`
+        The ID of the application that the entitlement belongs to.
+    type: :class:`EntitlementType`
+        The type of the entitlement.
+    consumed: :class:`bool`
+        Whether the entitlement has been consumed. Not applicable to app subscriptions so will typically be ``False``.
```

Comment:
```suggestion
    deleted: :class:`bool`
        Whether the entitlement has been deleted.
```
See https://github.com/discord/discord-api-docs/pull/6477